x/pairing/keeper/scores/score.go

```diff
@@ -188,30 +189,34 @@ func PickProviders(ctx sdk.Context, scores []*PairingScore, groupCount int, hash
 		return returnedProviders
 	}
 
-	for it := 0; it < groupCount; it++ {
-		hash := tendermintcrypto.Sha256(hashData) // TODO: we use cheaper algo for speed
-		bigIntHash := new(big.Int).SetBytes(hash)
-		uintHash := sdk.NewUintFromBigInt(bigIntHash)
-		modRes := uintHash.Mod(scoreSum)
+	sum256 := sha256.Sum256(hashData)
+	// Fold the SHA-256 hash into a 64-bit seed using bitwise XOR
+	var seed int64
+	for i := 0; i < len(sum256)/8; i++ {
+		seed ^= int64(binary.BigEndian.Uint64(sum256[i*8 : (i+1)*8]))
+	}
 
+	rand.Seed(seed)
```

Comment:
using the global `rand` directly is not safe in terms of determinism, e.g. if another go routine calls (now or in the future) it to get a random number at the wrong time. also, `rand.Seed` is deprecated in Go (for similar reasons). Instead, we should instantiate a dedicated instance using `rand.New()` - see below.

Also, Go does not guarantee that the PRNG driving the default source for `rand` will always provide the same output for a given seed - across different Go versions. Instead, we should provide our own source (PRNG) that will always be under our control. The [xorshift package](https://github.com/vpxyz/xorshift) offers a selection of such PRNGs. See [Xorshift wiki page](https://en.wikipedia.org/wiki/Xorshift) for details on the different options.

instead, we should instantiate a dedicated instance e.g.
```
   source := xorshift.XoroShiro128StarStar,NewSource(seed)
   rng := rand.New(source)
   ...
   value := rng.UInt63n(...)
```

Reply:
fixed 7944693a